WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt

```diff
@@ -1,171 +1,183 @@
-//            postSocialSharingModel = postSocialSharingModel,
-//            subscribeButtonLabel = "SHARE MORE",
-//            onSubscribeClick = onConnectProfilesClick,
-//        )
-//        assertThat(actual).isEqualTo(expected)
+        val shareLimit = ShareLimit.Enabled(
+            shareLimit = 10,
+            publicizedCount = 11,
+            sharedPostsCount = 12,
+            sharesRemaining = 13,
+        )
+        val shareMessage = "Message"
+        val subscribeButtonLabel = "SHARE MORE"
+        whenever(stringProvider.getString(R.string.post_settings_jetpack_social_subscribe_share_more))
+            .thenReturn(subscribeButtonLabel)
+        whenever(localProvider.getAppLocale()).thenReturn(Locale.US)
+
+        val allConnections = listOf(connection1, connection2)
+        val onConnectProfilesClick: () -> Unit = {}
+        val onShareMessageClick: () -> Unit = {}
+        val onConnectionClick: (connection: PostSocialConnection, enable: Boolean ) -> Unit = mock()
+
+        val actual = classToTest.mapLoaded(
+            connections = allConnections,
+            shareLimit = shareLimit,
+            onSubscribeClick = onConnectProfilesClick,
+            shareMessage = shareMessage,
+            onShareMessageClick = onShareMessageClick,
+            onConnectionClick = onConnectionClick,
+            isPostPublished = false
+        )
+        val expected = JetpackSocialUiState.Loaded(
+            jetpackSocialConnectionDataList = listOf(
+                JetpackSocialConnectionData(
+                    postSocialConnection = connection1,
+                    onConnectionClick = actual.jetpackSocialConnectionDataList[0].onConnectionClick,
+                    enabled = true
+                ),
+                JetpackSocialConnectionData(
+                    postSocialConnection = connection2,
+                    onConnectionClick = actual.jetpackSocialConnectionDataList[1].onConnectionClick,
+                    enabled = true
+                )
+            ),
+            showShareLimitUi = true,
+            isShareMessageEnabled = true,
+            shareMessage = shareMessage,
+            onShareMessageClick = onShareMessageClick,
+            subscribeButtonLabel = subscribeButtonLabel,
+            onSubscribeClick = onConnectProfilesClick
+        )
+
+        assertThat(actual).isEqualTo(expected)
+    }
+
+    @Test
+    fun `Loaded UI state should map the expected lambda for on onConnectionClick`() {
+        val shareLimit = ShareLimit.Enabled(
+            shareLimit = 10,
+            publicizedCount = 11,
+            sharedPostsCount = 12,
+            sharesRemaining = 13,
+        )
+        val shareMessage = "Message"
+        val subscribeButtonLabel = "SHARE MORE"
+        whenever(stringProvider.getString(R.string.post_settings_jetpack_social_subscribe_share_more))
+            .thenReturn(subscribeButtonLabel)
+        whenever(localProvider.getAppLocale())
```

Comment:
💡 Would it make sense to extract this part into a separate method, so it can be used by other tests mocking it instead of duplicating the mocking logic?

e.g.
```
private fun mockLocale() {
       whenever(localProvider.getAppLocale())
}
```